api/types/types.go

```diff
@@ -70,22 +70,24 @@ func NewAlert() *Alert {
 
 // Capture describes a capture API
 type Capture struct {
-	BasicResource  `yaml:",inline"`
-	GremlinQuery   string           `json:"GremlinQuery,omitempty" valid:"isGremlinExpr" yaml:"GremlinQuery"`
-	BPFFilter      string           `json:"BPFFilter,omitempty" valid:"isBPFFilter" yaml:"BPFFilter"`
-	Name           string           `json:"Name,omitempty" yaml:"Name"`
-	Description    string           `json:"Description,omitempty" yaml:"Description"`
-	Type           string           `json:"Type,omitempty" valid:"isValidCaptureType" yaml:"Type"`
-	Count          int              `json:"Count" yaml:"Count"`
-	PCAPSocket     string           `json:"PCAPSocket,omitempty" yaml:"PCAPSocket"`
-	Port           int              `json:"Port,omitempty" yaml:"Port"`
-	RawPacketLimit int              `json:"RawPacketLimit,omitempty" valid:"isValidRawPacketLimit" yaml:"RawPacketLimit"`
-	HeaderSize     int              `json:"HeaderSize,omitempty" valid:"isValidCaptureHeaderSize" yaml:"HeaderSize"`
-	ExtraTCPMetric bool             `json:"ExtraTCPMetric" yaml:"ExtraTCPMetric"`
-	IPDefrag       bool             `json:"IPDefrag" yaml:"IPDefrag"`
-	ReassembleTCP  bool             `json:"ReassembleTCP" yaml:"ReassembleTCP"`
-	LayerKeyMode   string           `json:"LayerKeyMode,omitempty" valid:"isValidLayerKeyMode" yaml:"LayerKeyMode"`
-	ExtraLayers    flow.ExtraLayers `json:"ExtraLayers,omitempty" yaml:"ExtraLayers"`
+	BasicResource   `yaml:",inline"`
+	GremlinQuery    string           `json:"GremlinQuery,omitempty" valid:"isGremlinExpr" yaml:"GremlinQuery"`
+	BPFFilter       string           `json:"BPFFilter,omitempty" valid:"isBPFFilter" yaml:"BPFFilter"`
+	Name            string           `json:"Name,omitempty" yaml:"Name"`
+	Description     string           `json:"Description,omitempty" yaml:"Description"`
+	Type            string           `json:"Type,omitempty" valid:"isValidCaptureType" yaml:"Type"`
+	Count           int              `json:"Count" yaml:"Count"`
+	PCAPSocket      string           `json:"PCAPSocket,omitempty" yaml:"PCAPSocket"`
+	Port            int              `json:"Port,omitempty" yaml:"Port"`
+	SamplingRate    uint32           `json:"SamplingRate" yaml:"SamplingRate"`
```

Comment:
You should add a validator in order to check that the value is > 0. Moreover you should add something here
https://github.com/skydive-project/skydive/blob/master/api/server/capture.go#L104
in order to set the default values. That is probably why the functionnal tests are not passing. SamplingRate is set to 0 by default.

Reply:
Done!
Added the default values inside the probe!
Thank You!